src/Dropdown.elm

```diff
@@ -67,33 +67,43 @@ type Selection item
         }
 
 -}
-type State
+type
+    State item
+    -- item type parameter is not currently used, but keeping it so if it does ever become needed it doesn't change the public api
```

Comment:
I added back the item type parameter into the public State type. Even though it's not currently used, I think it's good to keep it in so that if it does ever become needed it doesn't require a change to the public api. (It is then also more consistent with the Msg type which also takes an item type parameter).  As this change was in the last commit, I'm happy to revert just that commit if people would prefer it without an unused type parameter.